sentry/src/main/java/io/sentry/util/thread/MainThreadChecker.java

```diff
@@ -0,0 +1,28 @@
+package io.sentry.util.thread;
+
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Class that checks if a given thread is the Main/UI thread. The Main thread is denoted by the
+ * thread, where the `Sentry.init` method was called. If it was inited from a background thread,
+ * then this class is wrong, but it's a best effort.
+ *
+ * <p>We're gonna educate people through the docs.
+ */
+@ApiStatus.Internal
+public final class MainThreadChecker implements IMainThreadChecker {
+
+  private static final long mainThreadId = Thread.currentThread().getId();
```

Comment:
`m`: I think `mainThreadId` will only be set the first time accessing `MainThreadChecker. getInstance()` thus could be wrong / depending on the first caller.

Reply:
yep, but we are accessing it from Sentry.init only, so most of the time it should be correct (though not guaranteed that Sentry.init cannot be called from a background thread). There's also a doc comment to this class, check it out :) 

Or do you see a better way of identifying the main thread for Java desktop apps?

Reply:
Ah my bad, looks like I skimmed over that comment part. I don't know of any better way, so all good 👍